### PR TITLE
Added config enable=true/false for extensions

### PR DESCRIPTION
**Pull request by rafaeljigau**
Signed-off-by: Jigau Iosif Rafael <ijigau@cisco.com>

**Which issue does this PR fix**: #258

### CODE REVIEW

**Comment by adodon2go** [comment]
What happens in case sync is configured in disabled mode. Does it make sense to add a new test for this scenario?

**Comment by rafaeljigau** [comment]
If zot is started with sync feature not enabled, I dont think there is something to test.

If the config it's changed from enable true to enable false, as I understand @peusebiu's sync code, the sync won't stop, so more logic needs to be added I think.

File: pkg/api/controller.go [hunk 1]
```diff
@@ -253,7 +253,7 @@ func (c *Controller) InitImageStore() error {
 	}
 
 	// Enable extensions if extension config is provided
-	if c.Config.Extensions != nil && c.Config.Extensions.Sync != nil {
```
Comment (adodon2go) [inline pkg/api/controller.go:256]: I think we also need to check if sync feature is enabled [here](https://github.com/project-zot/zot/blob/main/pkg/api/routes.go#L1271) and [here](https://github.com/project-zot/zot/blob/main/pkg/api/routes.go#L1283)
Also check with Sebi if changes in pkg/cli/root.go are needed.

Comment (peusebiu) [reply pkg/api/controller.go:256]: Indeed, Alexei is right. 

Add a check here
https://github.com/project-zot/zot/blob/main/pkg/cli/root.go#L233

Comment (peusebiu) [reply pkg/api/controller.go:256]: Also here: https://github.com/project-zot/zot/blob/main/pkg/extensions/extensions.go#L103

**Pull request closed without merging**